### PR TITLE
[10.x] Add option to ignore case on string helper method before and after

### PR DESCRIPTION
**Pull request by bmckay959**
Added the option to ignore case on the `before` and `after` string helper. Set defaults to use the existing case sensitive functionality for backwards compatibility.

### CODE REVIEW

File: src/Illuminate/Support/Str.php [hunk 1]
```diff
@@ -83,8 +83,12 @@ public static function of($string)
      * @param  string  $search
      * @return string
      */
-    public static function after($subject, $search)
+    public static function after($subject, $search, $ignoreCase = false)
```
Comment (bmckay959) [inline src/Illuminate/Support/Str.php:86]: Default set for previous functionality

File: src/Illuminate/Support/Str.php [hunk 2]
```diff
@@ -142,13 +146,17 @@ public static function transliterate($string, $unknown = '?', $strict = false)
      * @param  string  $search
      * @return string
      */
-    public static function before($subject, $search)
+    public static function before($subject, $search, $ignoreCase = false)
```
Comment (bmckay959) [inline src/Illuminate/Support/Str.php:149]: Default set for previous functionality

File: tests/Support/SupportStrTest.php [hunk 3]
```diff
@@ -216,6 +216,25 @@ public function testStrBefore()
         $this->assertSame('han', Str::before('han0nah', '0'));
         $this->assertSame('han', Str::before('han0nah', 0));
         $this->assertSame('han', Str::before('han2nah', 2));
+        $this->assertSame('hannah', Str::before('hannah', 'NAH'));
```
Comment (bmckay959) [inline tests/Support/SupportStrTest.php:219]: Added a case sensitive to the previous functionality to show that it does not match

File: tests/Support/SupportStrTest.php [hunk 4]
```diff
@@ -271,6 +290,25 @@ public function testStrAfter()
         $this->assertSame('nah', Str::after('han0nah', '0'));
         $this->assertSame('nah', Str::after('han0nah', 0));
         $this->assertSame('nah', Str::after('han2nah', 2));
+        $this->assertSame('hannah', Str::after('hannah', 'HAN'));
```
Comment (bmckay959) [inline tests/Support/SupportStrTest.php:293]: Added a case sensitive test to the previous functionality to show that it does not match

**Comment by bmckay959** [comment]
Sorry for the mess of commits. I pushed up unintended code and had to clean it up.

File: src/Illuminate/Support/Str.php [hunk 5]
```diff
@@ -83,8 +83,12 @@ public static function of($string)
      * @param  string  $search
      * @return string
      */
-    public static function after($subject, $search)
+    public static function after($subject, $search, $ignoreCase = false)
     {
+        if ($ignoreCase) {
+            return $search === '' ? $subject : array_reverse(preg_split("/$search/i", $subject, 2))[0];
```
Comment (morloderex) [inline src/Illuminate/Support/Str.php:89]: @bmckay959 would it make sense to use preg_split here now that we are using regular expressions?

```suggestion
            $search = preg_qoute($search);
            return $search === '' ? $subject : array_reverse(preg_split("/$search/i", $subject, 2))[0];
```

Comment (bmckay959) [reply src/Illuminate/Support/Str.php:89]: I think you comment meant, "use preg_quote here..."?

If so, I think the answer is yes. I'm not super familiar with regex and didn't even know preg_quote existed, I'll try to build out a few test cases that need to have something escaped today and add it.

Thanks!

**Comment by taylorotwell** [comment]
Thanks for your pull request to Laravel!

Unfortunately, I'm going to delay merging this code for now. To preserve our ability to adequately maintain the framework, we need to be very careful regarding the amount of code we include. 

If possible, please consider releasing your code as a package so that the community can still take advantage of your contributions! 

If you feel absolutely certain that this code corrects a bug in the framework, please "**@**" mention me in a follow-up comment with further explanation so that GitHub will send me a notification of your response.

**Comment by bmckay959** [comment]
@taylorotwell I agree with the sentiment behind not bloating the framework. I would ask you to reconsider on this with the following scenario and if you still disagree, I can work around it, but it's too small a change to make into a package. Either way, grateful for you and the team.

See example below:

```php
$string = "LOOK FOR THIS! Additional text here...";
$search = "Look for this!";

// Note: ::contains allows ignoreCase as 3rd parameter
if(Str::contains($string, $search, true)) {

    // Currently, this does not find it and returns "LOOK FOR THIS! Additional text here..."
    $text = Str::after($string, $search);

    // Updates would allow this to return " Additional text here..."
    $text = Str::after($string, $search, true);
}
```

**Pull request closed without merging**